### PR TITLE
make all times Europe London

### LINKED ISSUE #55: Display times in Europe/London not UTC
Make sure all times are in the browser's time zone.

Note that all datetimes from API are in UTC

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Change all times to 'Europe/London'

- HH:MM are correct in graph
- forecast creation time is in 'Europe/London'
- target datetime is in 'Europe/London'

This was take around 09:14 Europe London time
Screenshot 2022-06-15 at 09 14 17

Fixes #55 

## How Has This Been Tested?

ran it locally, checked
- forecast creation time
- targte time on graph
- x lables
- label in hover on national gsp
- GSP plot looks correct too

Also CI check it builds

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: apps/nowcasting-app/components/utils.ts [hunk 1]
```diff
@@ -25,18 +25,34 @@ export const getTimeFromDate = (date: Date) => {
   return date.toTimeString().split(" ")[0].slice(0, -3);
 };
 export const formatISODateString = (date: string) => {
+  // Changes the ISO date string to Europe London time, and remove timezone and seconds
+
   const dateid = date?.slice(0, 16);
```
Comment (Alaatohamy) [inline apps/nowcasting-app/components/utils.ts:30]: can you delete it if it's not necessary anymore

**Review by Alaatohamy — APPROVED** [review]

File: apps/nowcasting-app/components/utils.ts [hunk 2]
```diff
@@ -25,18 +25,34 @@ export const getTimeFromDate = (date: Date) => {
...
+
+  const d = new Date(date);
+  const date_london_date_str = d.toLocaleDateString("en-GB", { timeZone: "Europe/London" });
+  // make string in DD/MM/YYYY format
+  const date_london_time_str = d
+    .toLocaleTimeString("en-GB", { timeZone: "Europe/London" })
+    .slice(0, 5);
+  // make string in HH:MM format
+  const year = date_london_date_str.slice(6, 10);
+  const month = date_london_date_str.slice(3, 5);
```
Comment (dlinah) [inline apps/nowcasting-app/components/utils.ts:40]: this function get used to convert all the dates form the api , so when i ran it locally i noticed that clicking on the chart causes the page to freeze . 
this function need to be fast .
i'm looking for a way to convert it without using javascript Date class

**Review by dlinah — COMMENTED** [review]
LGTM ,

**Review by dlinah — APPROVED** [review]
@peterdudfield did a small change for the performance 🙈 , fell free to merge if your happy with it

File: apps/nowcasting-app/components/utils.ts (hunk 1 above)
Comment (peterdudfield) [reply apps/nowcasting-app/components/utils.ts:30]: Thanks, i think this is now sorted, this is now needed in 'formatISODateString'

**Comment by peterdudfield** [comment]
> @peterdudfield did a small change for the performance 🙈 , fell free to merge if your happy with it

Unfortunately those changes now mean the HH:MM are now in UTC on the graph

**Pull request merged**